src/generator/AutoRest.CSharp.Azure.Fluent/TransformerCsaf.cs

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// 
+
+using System.Linq;
+using AutoRest.Core;
+using AutoRest.Core.Model;
+using AutoRest.CSharp.Azure.Fluent.Model;
+using AutoRest.CSharp.Azure.Model;
+using AutoRest.CSharp.Model;
+using AutoRest.Extensions.Azure;
+
+namespace AutoRest.CSharp.Azure.Fluent
+{
+    public class TransformerCsaf : TransformerCsa, ITransformer<CodeModelCsaf>
+    {
+        CodeModelCsaf ITransformer<CodeModelCsaf>.TransformCodeModel(CodeModel cs)
+        {
+            var codeModel = cs as CodeModelCsaf;
+
+            // Do parameter transformations
+            TransformParameters(codeModel);
+
+            // todo: these should be turned into individual transformers
+            AzureExtensions.NormalizeAzureClientModel(codeModel);
+
+            // Fluent Specific stuff.
+            NormalizeResourceTypes(codeModel);
+            NormalizeTopLevelTypes(codeModel);
+            NormalizeModelProperties(codeModel);
+
+
+            NormalizePaginatedMethods(codeModel);
+            NormalizeODataMethods(codeModel);
+
+            return codeModel;
+        }
+
+        /// <summary>
+        ///     A type-specific method for code model tranformation.
+        ///     Note: This is the method you want to override.
+        /// </summary>
+        /// <param name="codeModel"></param>
+        /// <returns></returns>
+        public override CodeModelCs TransformCodeModel(CodeModel codeModel)
+        {
+            return ((ITransformer<CodeModelCsaf>) this).TransformCodeModel(codeModel);
+        }
+
+        public void NormalizeResourceTypes(CodeModelCsaf codeModel)
+        {
+            if (codeModel != null)
+            {
+                foreach (var model in codeModel.ModelTypes)
+                {
+                    if ((model.BaseModelType != null) && (model.BaseModelType.Extensions != null) &&
+                        model.BaseModelType.Extensions.ContainsKey(AzureExtensions.AzureResourceExtension) &&
+                        (bool) model.BaseModelType.Extensions[AzureExtensions.AzureResourceExtension])
```

Comment:
I can't see that necessarily being true (again, not my code) ... but I think we can wait until we scrub the way extensions are done.